tests/e2e-prometheuscr/create-sm-prometheus-exporters/02-assert.yaml

```diff
@@ -0,0 +1,41 @@
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  labels:
+    app.kubernetes.io/instance: create-sm-prometheus.simplest
+    app.kubernetes.io/managed-by: opentelemetry-operator
+    app.kubernetes.io/name: simplest-collector
+  name: simplest-collector
+  namespace: create-sm-prometheus
+spec:
+  endpoints:
+  - port: monitoring
+  - port: prometheus-prod
+  namespaceSelector:
+    matchNames:
+    - create-sm-prometheus
+  selector:
+    matchLabels:
+      app.kubernetes.io/managed-by: opentelemetry-operator
```

Comment:
It seems that this will match all OTEL CRs in a namespace. I think it should be more specific.

Reply:
Good catch! I added some changes to ensure this not happens